akka-docs/src/main/paradox/stream/operators/Compression/gzip.md

```diff
@@ -0,0 +1,30 @@
+# Compression.gzip
+
+Creates a flow that gzip-compresses a stream of ByteStrings.  
+
+@ref[Compression operators](../index.md#compression-operators)
+
+## Signature
+
+@apidoc[Compression.gzip](stream.*.Compression$) { scala="#gzip:akka.stream.scaladsl.Flow[akka.util.ByteString,akka.util.ByteString,akka.NotUsed]" java="#gzip()" }
+
+## Description
+
+Creates a flow that gzip-compresses a stream of ByteStrings. Note that the compressor
+will SYNC_FLUSH after every @apidoc[akka.util.ByteString] so that it is guaranteed that every @apidoc[akka.util.ByteString]
+coming out of the flow can be fully decompressed without waiting for additional data. This may
+come at a compression performance cost for very small chunks.
+
+Use the overload with the `level:Int` parameter to control the compression level.
```

Comment:
Scala vs Java syntax in the snippet still.

Reply:
Oh, good catch.